kernel/src/process/process_vm/mod.rs

```diff
@@ -160,10 +163,24 @@ impl ProcessVm {
         &self.heap
     }
 
-    /// Clears existing mappings and then maps stack and heap vmo.
-    pub(super) fn clear_and_map(&self) {
+    /// Clears existing mappings and then maps the heap VMO to the current VMAR.
+    pub fn clear_and_map(&self) {
         let root_vmar = self.lock_root_vmar();
         root_vmar.get().clear().unwrap();
         self.heap.alloc_and_map_vm(&root_vmar.get()).unwrap();
     }
 }
+
+/// Renews the [`ProcessVm`] of the current process and then maps the heap VMO to the new VMAR.
+pub fn renew_and_map(ctx: &Context) {
```

Comment:
If it's not an associated method of `ProcessVm`, I think it should be renamed to `renew_vm_and_map` to clarify what it intends to renew.